pipeline/frontend/yaml/constraint/constraint.go

```diff
@@ -47,9 +52,82 @@ type (
 	}
 )
 
+func (when *When) IsEmpty() bool {
+	return len(when.Constraints) == 0
+}
+
+// Returns true if at least one of the internal constraints is true.
+func (when *When) Match(metadata frontend.Metadata) bool {
+	for _, c := range when.Constraints {
+		if c.Match(metadata) {
+			return true
+		}
+	}
+	return when.IsEmpty()
+}
+
+func (when *When) IncludesStatus(status string) bool {
+	for _, c := range when.Constraints {
+		if c.Status.Includes(status) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (when *When) ExcludesStatus(status string) bool {
+	for _, c := range when.Constraints {
+		if !c.Status.Excludes(status) {
+			return false
+		}
+	}
+
+	return len(when.Constraints) > 0
+}
+
+// False if (any) non local
+func (when *When) IsLocal() bool {
+	for _, c := range when.Constraints {
+		if !c.Local.Bool() {
+			return false
+		}
+	}
+	return true
+}
+
+func (when *When) UnmarshalYAML(value *yaml.Node) error {
+	unmarshelAsList := func() error {
```

Comment:
https://github.com/woodpecker-ci/woodpecker/pull/826/files#diff-999f3a28c197f25cb56892833339acc43cef0691e5622617dcf4d26c31cf7cb8R68

We should switch case and not try-err

Reply:
#1112